src/classes/RD2_EntryFormController_TEST.cls

```diff
@@ -375,28 +266,19 @@ private with sharing class RD2_EntryFormController_TEST {
         RD2_EnablementService_TEST.setRecurringDonations2Enabled();
         PS_IntegrationService.setConfiguration(PS_IntegrationServiceConfig_TEST.testConfig);
 
-        npe03__Recurring_Donation__c rd = TEST_RecurringDonationBuilder.constructEnhancedBuilder()
-            .withDefaultValues()
-            .withContact(getContact().Id)
-            .withAmount(100)
-            .withDateEstablished(START_DATE)
-            .withStartDate(START_DATE)
-            .withDayOfMonth('1')
-            .withCommitmentId(TEMP_COMMITMENT_ID)
-            .build();
-        insert rd;
+        npe03__Recurring_Donation__c rd = buildRecurringDonation();
 
         Test.startTest();
         UTIL_Http_TEST.mockBadRequestCalloutResponse();
 
-        String jsonResponse = RD2_EntryFormController.createCommitment(rd.Id, PAYMENT_METHOD_TOKEN);
+        String jsonResponse = RD2_EntryFormController.handleCommitment(JSON.serialize(rd), PAYMENT_METHOD_TOKEN);
         Test.stopTest();
 
         List<Error__c> errors = errorGateway.getRecords();
         System.assertEquals(1, errors.size(),
             'An error record should be created for an error response.');
-        System.assertEquals(rd.Id, errors[0].Related_Record_Id__c,
-            'The error record should contain RD record Id.');
+        System.assertEquals(rd.npe03__Contact__c, errors[0].Related_Record_Id__c,
```

Comment:
The change has been made that a donor Id is set as the record Id in the error log when commitment is created but Recurring Donation creation failed.